File: SampleDotnet/Microservices/OnlineMovieStore/src/Web.MovieStore/ClientApp/src/models/responses/movies/movie.dto.ts
import { MovieCategoryDto } from "./movie-category.dto";
import { MovieDirectorDto } from "./movie-director.dto";
import { MovieWriterDto } from "./movie-writer.dto";
import { RatingDto } from "./rating.dto";

export interface MovieDto {
  categories: MovieCategoryDto[];
  moviedirectors: MovieDirectorDto[];
  moviewriters: MovieWriterDto[];
  id: string;
  title: string;
  runtimeminutes: number;
  startyear: number;
  description: string;
  rating: RatingDto;
  type: string;
  usdprice: number;
  itemdatabase: string;
}

File: SampleDotnet/Microservices/OnlineMovieStore/src/MovieService/Movie.API/Models/Dtos/MovieDto.cs
using Newtonsoft.Json;
using SampleProject.Movie.Database.Enums;

namespace SampleProject.Movie.API.Models.Dtos
{
    public class MovieDto
    {
        [JsonProperty("categories")]
        public List<MovieCategoryDto> Categories { get; set; }

        [JsonProperty("moviedirectors")]
        public List<MovieDirectorDto> MovieDirectors { get; set; }

        [JsonProperty("moviewriters")]
        public List<MovieWriterDto> MovieWriters { get; set; }

        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("runtimeminutes")]
        public int RuntimeMinutes { get; set; }

        [JsonProperty("startyear")]
        public int StartYear { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("rating")]
        public RatingDto Rating { get; set; }

        [JsonProperty("type")]
        public MovieType Type { get; set; }

        [JsonProperty("usdprice")]
        public decimal UsdPrice { get; set; }

        [JsonProperty("itemdatabase")]
        public string ItemDatabase { get; set; }
    }
}